src/components/ConfigEditor.tsx

```diff
@@ -1,135 +1,28 @@
   DataSourcePluginOptionsEditorProps,
-  onUpdateDatasourceJsonDataOption,
   onUpdateDatasourceJsonDataOptionSelect,
 } from '@grafana/data';
-import { Field, FieldSet, Input, RadioButtonGroup, Select } from '@grafana/ui';
+import { Field, FieldSet, Select } from '@grafana/ui';
+import { AuthConfig, GOOGLE_AUTH_TYPE_OPTIONS } from '@grafana/google-sdk';
 
 import React from 'react';
-import { JWTConfigEditor } from './JWTConfigEditor';
-import { JWTForm } from './JWTForm';
 import { ConfigurationHelp } from './/ConfigurationHelp';
-import { GOOGLE_AUTH_TYPE_OPTIONS, PROCESSING_LOCATIONS } from '../constants';
-import { BigQueryOptions, BigQuerySecureJsonData, GoogleAuthType } from '../types';
+import { PROCESSING_LOCATIONS } from '../constants';
+import { BigQueryOptions, BigQuerySecureJsonData } from '../types';
 
 export type BigQueryConfigEditorProps = DataSourcePluginOptionsEditorProps<BigQueryOptions, BigQuerySecureJsonData>;
 
-const isJWTAuth = (authenticationType: GoogleAuthType): boolean => {
-  return authenticationType === GoogleAuthType.JWT || authenticationType === undefined;
-};
-
 export const BigQueryConfigEditor: React.FC<BigQueryConfigEditorProps> = (props) => {
-  const { options, onOptionsChange } = props;
-  const { jsonData, secureJsonFields, secureJsonData } = options;
-  if (!jsonData.authenticationType) {
-    jsonData.authenticationType = GoogleAuthType.JWT;
-  }
-
-  const [jwtAuth, setJWTAuth] = React.useState<boolean>(isJWTAuth(jsonData.authenticationType));
-
-  const getJTWConfig = (): boolean =>
-    Boolean(
-      jsonData.clientEmail &&
-        jsonData.defaultProject &&
-        jsonData.tokenUri &&
-        ((secureJsonFields && secureJsonFields.privateKey) || jsonData.privateKeyPath)
-    );
-
-  const [configEditorVisible, setConfigEditorVisible] = React.useState<boolean>(getJTWConfig());
-
-  const showConfigEditor = (): void => {
-    setConfigEditorVisible(true);
-  };
-
-  const onAuthTypeChange = (authenticationType: GoogleAuthType) => {
-    setConfigEditorVisible(getJTWConfig());
-    onResetApiKey({ authenticationType });
-    setJWTAuth(isJWTAuth(authenticationType));
-  };
-
-  const onResetApiKey = (jsonData?: Partial<BigQueryOptions>) => {
-    const nextSecureJsonData = { ...secureJsonData };
-    const nextJsonData = !jsonData ? { ...options.jsonData } : { ...options.jsonData, ...jsonData };
-
-    delete nextJsonData.clientEmail;
-    delete nextJsonData.defaultProject;
-    delete nextJsonData.tokenUri;
-    delete nextJsonData.privateKeyPath;
-    delete nextSecureJsonData.privateKey;
-
-    setJWTAuth(true);
-    setConfigEditorVisible(false);
-    onOptionsChange({
-      ...options,
-      secureJsonFields: { ...options.secureJsonFields, privateKey: false },
-      secureJsonData: nextSecureJsonData,
-      jsonData: nextJsonData,
-    });
-  };
-
-  const onJWTFormChange = (key: keyof BigQueryOptions) => onUpdateDatasourceJsonDataOption(props, key);
+  const { options } = props;
```

Comment:
Will there always be an `options` and `jsonData` object? If not, this destructuring will error out. If so, maybe consider `const x = props?.options?.jsonData`?